src/tests/GfxBatchHbaoTest.cpp

```diff
@@ -438,53 +503,130 @@ void GfxBatchHbaoTest::testFlippedHBAOData(const TestDataType& data,
+                                         Mn::Matrix4 projMatrix) {
+  if ((data.config.flags() & esp::gfx_batch::HbaoFlag::LayeredImageLoadStore) &&
+      !(
+#ifdef MAGNUM_TARGET_GLES
+          Mn::GL::Context::current().isVersionSupported(
+              Mn::GL::Version::GLES310)
+#else
+          Mn::GL::Context::current()
+              .isExtensionSupported<
+                  Mn::GL::Extensions::ARB::shader_image_load_store>()
+#endif
+              ))
+    CORRADE_SKIP("Image load/store not supported");
+
+  // Scale image size to use for benchmark uniformly so projection matrix aspect
+  // ration not affected.
+  Mn::Vector2i BenchImageSize = Size * 4;
+  const size_t arraySize = BenchImageSize.x() * BenchImageSize.y() * 8;
+
+  // For benchmarks source color and depth are just empty white images
+
+  Cr::Containers::Array<char> depthData{Cr::DirectInit, arraySize, '\xff'};
+  Mn::Trade::ImageData2D depth{Mn::PixelFormat::Depth32F, BenchImageSize,
+                               std::move(depthData)};
+
+  Mn::GL::Texture2D inputDepthTexture;
+  inputDepthTexture
+      .setStorage(1, Mn::GL::TextureFormat::DepthComponent32F, depth.size())
+      .setSubImage(0, {}, depth);
+
+  Cr::Containers::Array<char> colorData{Cr::DirectInit, arraySize, '\xff'};
+  Mn::Trade::ImageData2D color{Mn::PixelFormat::RGBA8Unorm, BenchImageSize,
+                               std::move(colorData)};
+  Mn::GL::Texture2D outputColorTexture;
+  outputColorTexture.setStorage(1, Mn::GL::TextureFormat::RGBA8, color.size())
+      .setSubImage(0, {}, color);
+
+  Cr::Containers::Array<char> resColorData{Cr::DirectInit, arraySize, '\xff'};
+  Mn::Trade::ImageData2D resultImage{Mn::PixelFormat::RGBA8Unorm,
+                                     BenchImageSize, std::move(resColorData)};
+
+  Mn::GL::Framebuffer output{{{}, BenchImageSize}};
+  output.attachTexture(Mn::GL::Framebuffer::ColorAttachment{0},
+                       outputColorTexture, 0);
+
+  /* No clear, that would kill the base image */
+
+  MAGNUM_VERIFY_NO_GL_ERROR();
+
+  auto config{data.config};
+  esp::gfx_batch::Hbao hbao{config.setSize(BenchImageSize)};
+  MAGNUM_VERIFY_NO_GL_ERROR();
+  // Call once to compile the shaders
+  hbao.drawEffect(projMatrix, !data.classic, inputDepthTexture, output);
+
+  // benchmark projection drawing for both classic and cache-aware algorithms
+  CORRADE_BENCHMARK(16) {
+    hbao.drawEffect(projMatrix, !data.classic, inputDepthTexture, output);
+  }
+
+  MAGNUM_VERIFY_NO_GL_ERROR();
+
+  CORRADE_COMPARE_WITH(
+      output.read({{}, BenchImageSize}, {Mn::PixelFormat::RGBA8Unorm}),
+      resultImage,
+      (Mn::DebugTools::CompareImage{data.maxThreshold, data.meanThreshold}));
+
+}  // GfxBatchHbaoTest::testHBAOData
+
+void GfxBatchHbaoTest::benchmarkPerspective() {
+  auto&& data = BenchData[testCaseInstanceId()];
+  setTestCaseDescription(
+      Cr::Utility::format("{}, perspective benchmark.", data.name));
```

Comment:
```suggestion
      Cr::Utility::format("{}, perspective", data.name));
```

Redundant, same below.